labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -77,7 +106,11 @@ object RecursionPatternMatchingExercise {
    * List(List(1), List('A, 'B, 'C), List('a, 'b)) -> List(List(1, 'A, 'a))
    */
   def zipMultipleWithDifferentSize(in: List[List[_]]): List[List[_]] = {
-    error("fix me")
+    in match {
+      case Nil => error("This operation does't make sense for no lists")
+      case x :: Nil => x.map(List(_))
+      case x :: rest => x.zip(zipMultiple(rest)).map(tupled(_::_))
+    }
```

Comment:
For consistency, should be zipMultipleWithDifferentSize but you could have also just called zipMultiple in the first place :+1: